### ISSUE TITLE
Have to press back twice, if Add transaction is exited with autocomplete open

### ISSUE THREAD

**Issue by bradyt** [bradyt/cone#13]
v0.1.5 introduced exception, setState() called after dispose()

This issue was introduced with the new account autocompletion feature. If user goes to "Add transaction" page, enters a character to prompt autocomplete suggestions, with completion suggestions showing, then presses back button, then settings button (or plus button), then back button again, and the following trace, then back button finally works second time.

<details><summary>log</summary>

```
I/flutter ( 4971): ══╡ EXCEPTION CAUGHT BY FOUNDATION LIBRARY ╞════════════════════════════════════════════════════════
I/flutter ( 4971): The following assertion was thrown while dispatching notifications for TextEditingController:
I/flutter ( 4971): setState() called after dispose(): AddTransactionState#dc1f0(lifecycle state: defunct, not mounted)
I/flutter ( 4971): This error happens if you call setState() on a State object for a widget that no longer appears in
I/flutter ( 4971): the widget tree (e.g., whose parent widget no longer includes the widget in its build). This error
I/flutter ( 4971): can occur when code calls setState() from a timer or an animation callback.
I/flutter ( 4971): The preferred solution is to cancel the timer or stop listening to the animation in the dispose()
I/flutter ( 4971): callback. Another solution is to check the "mounted" property of this object before calling
I/flutter ( 4971): setState() to ensure the object is still in the tree.
I/flutter ( 4971): This error might indicate a memory leak if setState() is being called because another object is
I/flutter ( 4971): retaining a reference to this State object after it has been removed from the tree. To avoid memory
I/flutter ( 4971): leaks, consider breaking the reference to this object during dispose().
I/flutter ( 4971): 
I/flutter ( 4971): When the exception was thrown, this was the stack:
I/flutter ( 4971): #0      State.setState.<anonymous closure> (package:flutter/src/widgets/framework.dart:1105:9)
I/flutter ( 4971): #1      State.setState (package:flutter/src/widgets/framework.dart:1140:6)
I/flutter ( 4971): #2      AddTransactionState.didChangeDependencies.<anonymous closure> (package:cone/src/add_transaction.dart:55:47)
I/flutter ( 4971): #3      ChangeNotifier.notifyListeners (package:flutter/src/foundation/change_notifier.dart:206:21)
I/flutter ( 4971): #4      ValueNotifier.value= (package:flutter/src/foundation/change_notifier.dart:273:5)
I/flutter ( 4971): #5      TextEditingController.text= (package:flutter/src/widgets/editable_text.dart:146:5)
I/flutter ( 4971): #6      PostingWidget.build.<anonymous closure> (package:cone/src/posting_widget.dart:66:33)
I/flutter ( 4971): #7      _TypeAheadFieldState._initOverlayEntry.<anonymous closure>.<anonymous closure> (package:flutter_typeahead/flutter_typeahead.dart)
I/flutter ( 4971): #8      _SuggestionsListState.createSuggestionsWidget.<anonymous closure>.<anonymous closure> (package:flutter_typeahead/flutter_typeahead.dart)
I/flutter ( 4971): #9      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:635:14)
I/flutter ( 4971): #10     _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:711:32)
I/flutter ( 4971): #11     GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:182:24)
I/flutter ( 4971): #12     TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:365:11)
I/flutter ( 4971): #13     TapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:275:7)
I/flutter ( 4971): #14     PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:455:9)
I/flutter ( 4971): #15     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:75:13)
I/flutter ( 4971): #16     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:102:11)
I/flutter ( 4971): #17     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:218:19)
I/flutter ( 4971): #18     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:198:22)
I/flutter ( 4971): #19     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:156:7)
I/flutter ( 4971): #20     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:102:7)
I/flutter ( 4971): #21     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:86:7)
I/flutter ( 4971): #25     _invoke1 (dart:ui/hooks.dart:250:10)
I/flutter ( 4971): #26     _dispatchPointerDataPacket (dart:ui/hooks.dart:159:5)
I/flutter ( 4971): (elided 3 frames from package dart:async)
I/flutter ( 4971): 
I/flutter ( 4971): The TextEditingController sending notification was:
I/flutter ( 4971):   TextEditingController#9edf6(TextEditingValue(text: ┤a├, selection: TextSelection(baseOffset: -1,
I/flutter ( 4971):   extentOffset: -1, affinity: TextAffinity.downstream, isDirectional: false), composing:
I/flutter ( 4971):   TextRange(start: -1, end: -1)))
I/flutter ( 4971): ════════════════════════════════════════════════════════════════════════════════════════════════════
I/flutter ( 4971): Another exception was thrown: Looking up a deactivated widget's ancestor is unsafe.
```

</detail>

**Comment by bradyt** [comment]
I have some changes that get avoid the exception, but still have to press back button twice.

**Comment by bradyt** [comment]
Fixed in v0.2.1+9 at https://github.com/bradyt/cone/commit/24d347ca7dd20b9f819747f1e255dd5cafdad16f.
